fedgov-cv-lib-asn1c/src/main/java/wrapper.c

```diff
@@ -142,6 +142,42 @@ JNIEXPORT jbyteArray JNICALL Java_gov_usdot_cv_mapencoder_Encoder_encodeMap(JNIE
 			intersectionId = (long)geomId;
 			intersection->id.id = intersectionId;
 
+		
+			RoadAuthorityID_t roadAuthorityID;
+
+			// Check if full RAID exists
+			jmethodID isFullRdAuthIDExists = (*env)->GetMethodID(env, intersectionClass, "isFullRdAuthIDExists", "()Z");
+			jboolean fullRdAuthIDExists = (*env)->CallBooleanMethod(env, intersectionObj, isFullRdAuthIDExists);
+
+			
+			if (fullRdAuthIDExists)
+			{
+				// Get full RAID
+				jmethodID getfullRdAuthID = (*env)->GetMethodID(env, intersectionClass, "getFullRdAuthID", "()Ljava/lang/String;");
+				jstring fullRdAuthID = (*env)->CallObjectMethod(env, intersectionObj, getFullRdAuthID);
+
+				roadAuthorityID.present = RoadAuthorityID_PR_fullRdAuthID;
+				roadAuthorityID.choice.fullRdAuthID = (RoadAuthorityID_PR_fullRdAuthID)fullRdAuthID;
+			} else {
```

Comment:
If fullRdAuthIDExists is false, check for isRelRdAuthIDExists too. If this is false too, then set roadAuthorityID.present to RoadAuthorityID_PR_NOTHING;